x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/methods/create_custom_rule.ts

```diff
@@ -26,5 +28,15 @@ export const createCustomRule = async (
     data: internalRule,
   });
 
-  return rule;
+  /* Trying to convert the rule to a RuleResponse object */
+  const parseResult = RuleResponse.safeParse(transform(rule));
+
+  if (!parseResult.success) {
+    throw new RuleResponseValidationError({
+      message: stringifyZodError(parseResult.error),
+      ruleId: rule.params.ruleId,
+    });
+  }
+
+  return parseResult.data;
```

Comment:
I'm not suggesting to do it in this PR to avoid preliminary generalization, but there's a chance that we'll need to extract this piece into a reusable "transform and validate" type of function. We could consider doing this when we start refactoring converters.

Reply:
Good idea. Added as a todo to the [issue](https://github.com/elastic/kibana/issues/184364).